### ISSUE TITLE
filter op waarde naam werkt niet

### ISSUE THREAD

**Issue by WimLo** [libis/archiefpunt-data-api#3]
query: `/agenten?filter[naam.waarde][contains]=Sint-Catharina&include=naam`

Dit geeft toch alle agents als resultaat.

**Comment by WimLo** [comment]
Ook hier werkt filter niet:

`/archieven?filter[identificatienummer.waarde][contains]=842474&include=identificatienummer`

**Comment by mehmetc** [comment]
Het lijkt erop dat er een fout in de conversie zit.
Volgende is een werkende query maar ik vermoed dat je dit niet wil. Je krijgt hier alle archieven en enkel data voor het identificatienummer waar de waardes 75 bevatten
/archieven?filter[identificatienummer.waarde][contains]=75&include=identificatienummer

De query die archieven beperkt is de volgende. Hier zou niet 75 mogen staan maar eerder de uuid van het identificatienummer object. 
/archieven?filter[identificatienummer_id][eq]=75

Ik laat de case open tot we de data convertie opgelost hebben

**Comment by WimLo** [comment]
Ter info

M.b.t. deze query `/agenten?filter[naam.waarde][contains]=Sint-Catharina&include=naam`

In de JSON-output krijg ik onder de key `included` inderdaad enkel de naam van het beoogde record terug. Maar onder de key `data` krijg ik gewoon een lijst van alle agenten.

```
"included": [{
	"id": "01cfa653-8ca7-4c41-a493-92b671c4f95a",
	"type": "namen",
	"attributes": {
		"waarde": "Parochie Sint-Catharina, Steenokkerzeel (Humelgem)"
	},
	"relationships": {
		"type_naam": {
			"links": {
				"related": "/type_namen/3043706c-3a5a-5fd7-8afd-283a3770f9cf"
			}
		}
	}
}]
```

**Comment by mehmetc** [comment]
Dat is wat ik bedoel. Je doet een "deep query" alle agenten en enkel data(included) waar in de naam entiteit 'Sint-Catharina' in het waarde attribute zit. Ik geef toe dat het een vreemde manier van denken is maar dat zijn nu graphs. Je kan beperken op de verzameling agenten en op de verzameling namen maar je kan de doorsnede niet opvragen(geen join doen). Dat is ook de reden waarom ze deze dingen vaak grafisch voorstellen. Dus als naam.waarde via agenten vaak gebruikt wordt moeten we die op niveau van de agenten beperken door een extra index te definieren.

**Comment by WimLo** [comment]
Bedankt voor de info. Begrijp ik het dan juist dat als ik een JSON wil verkrijgen die gewoon bestaat uit een lijst van alle agenten met 'Sint' in de naam, inclusief hun identificatienummer, dat ik dan een script op de output van volgende API-query moet laten lopen? `/agenten?filter[naam.waarde][contains]=Sint&include=naam,identificatie` 

Dit script zou dan over de volledige set van 32932 agenten itereren en vervolgens koppelen met de gerelateerde naam en identificatienummer indien er een koppeling is?

**Comment by mehmetc** [comment]
Yep, het is echt zo absurd als het klinkt. GraphQL heeft de zelfde “features”. Ergens ook wel logisch als je je DB opent voor iedereen. We kunnen dit op 2 manieren oplossen een construct query dat de data teruggeeft zoals jij ze wil. Dit moeten we voor de web developers ook gaan doen, btw. Of extra indexen.
